lib/inspec/cached_fetcher.rb

```diff
@@ -39,12 +39,21 @@ def cache_key
     end
 
     def fetch
-      if cache.exists?(cache_key)
+      if cache.exists?(cache_key) && cache.locked?(cache_key)
+        Inspec::Log.debug "Waiting for lock to be released on the cache dir ...."
+        sleep 1
+        fetch
+      elsif cache.exists?(cache_key) && !cache.locked?(cache_key)
         Inspec::Log.debug "Using cached dependency for #{target}"
         [cache.prefered_entry_for(cache_key), false]
       else
         Inspec::Log.debug "Dependency does not exist in the cache #{target}"
-        fetcher.fetch(cache.base_path_for(fetcher.cache_key))
+        cache.lock(cache_key) if fetcher.class == Inspec::Fetcher::Git
```

Comment:
For archived formatted cache directory the inspec parallel exec does not break and doesn't need locking. Git fetcher is the only one which does not use archived formatted caching and requires the lock. As it is currently only required by Git fetcher I feel this whole logic of cache locking should go inside Git fetcher. Need to understand if there are different opinions on this.